File: tests/fixtures/header_injection/javascript/unsafe_set_header.js
// Unsafe: Express `res.setHeader` receives a value built from req.query.
// HEADER_INJECTION fires on the value argument.
function handler(req, res) {
    const lang = req.query.lang;
    res.setHeader('X-Lang', lang);
    res.end();
}

module.exports = handler;

File: tests/fixtures/header_injection/java/UnsafeSetHeader.java
// Unsafe: HttpServletResponse.setHeader receives a value built from a
// request parameter.  HEADER_INJECTION fires on the value argument.
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public class UnsafeSetHeader {
    public void handle(HttpServletRequest req, HttpServletResponse res) {
        String lang = req.getParameter("lang");
        res.setHeader("X-Lang", lang);
    }
}
